Board.java

```diff
@@ -1,42 +1,63 @@
 public class Board {
     private String state;
-    private int move;
+    private char lastTurn = 'o';
+    private int width;
 
-    public Board(String state, int move) {
+    public Board(String state) {
         this.state = state;
-        this.move = move;
+        width = state.length();
     }
 
     public String getState() {
         return this.state;
     }
 
-    public int getMove() {
-        return this.move;
-    }
-
-
     public void endGame(char winner) {
         System.out.print(winner + " won!");
-        this.state = "---------";
+        state = "---------";
     }
 
-    public boolean checkForWin(char ply) {
-        int count = 0;
-        for (int i=0; i < 3; i++) {
-               if (this.state.charAt(i) == ply) count += 1;
-            if (count==3) return true;
-        }
-        int count2;
-        for (int i=0; i < 3; i++) {
-            if (this.state.charAt(i+2) == ply) count += 1;
-            if (count==3) return true;
+    public Board playerMakesMove(char ply, int pos, int move) {
+        StringBuilder currentBoard = new StringBuilder(state);
+        if (currentBoard.charAt(pos) == '-' && checkTurn(ply)) {
+            currentBoard.setCharAt(pos, ply);
+            lastTurn = ply;
+            Board newBoard = new Board(currentBoard.toString());
+            if (newBoard.checkForWin(ply)) {
+                return null;
```

Comment:
Do you really need to return null if they have won? Why not just return the same board?